controllers/opentelemetrycollector_controller.go

```diff
@@ -63,10 +67,55 @@ type Params struct {
 	Config   config.Config
 }
 
+func (r *OpenTelemetryCollectorReconciler) onPlatformChange() error {
```

Comment:
I find this method hard to read a bit. 

It would be cleaner if the function just got the platform and if the route task is missing it would add it or removed it.

Reply:
I understand. Unfortunately, the platform cannot be transferred. Because the registered callbacks have the signature `func() error`.

Would it help to define two sub methods?
e.g.
```golang
func (r *OpenTelemetryCollectorReconciler) onPlatformChange() error {
	plt := r.config.Platform()
	if err := r.addRouteReconciler(plt); err != nil {
		return err
	}
	return r.removeRouteReconciler(plt)
}
```